rebench/executor.py

```diff
@@ -140,14 +146,17 @@ class RandomScheduler(RunScheduler):
 
     def _process_remaining_runs(self, runs):
         task_list = list(runs)
-
         while task_list:
             run = random.choice(task_list)
             try:
                 completed = self._executor.execute_run(run)
                 self._indicate_progress(completed, run)
                 if completed:
                     task_list.remove(run)
+                if run.benchmark.suite.executor.binary_missing:
+                    task_list = deque(self._executor.without_missing_binaries(
+                        run.benchmark.suite.executor, task_list))
+
```

Comment:
Same as above, here you want to indent this to be in the `is complete` branch for the same reasons. The binary can only be missing once the run completed.